python/pyarrow/table.pxi

```diff
@@ -5334,7 +5334,7 @@ class TableGroupBy:
         Parameters
         ----------
         aggregations : list[tuple(str, str)] or \
-list[tuple(str, str, FunctionOptions)]
+list[tuple(str|list[str]|tuple(str*), str, FunctionOptions)]
```

Comment:
Basically, IIUC the "only" thing that this PR is adding is that the target column name (first argument of the tuple) can be a list of names instead of a single name (string). API wise I think that is reasonable? (I am not sure alternatives would be better, like not passing any column name and inferring that from the length of the tuple ..).

(one alternative would be to make it more structured, and provide a some namedtuple or small class to define an "aggregation", and then this argument would be a list of such objects. But that won't necessarily make it easier to use, and we probably want to keep supporting the plain tuple as well)

But personally I do find the above line explaining the type completely unreadable, though. I would simplify to just "list of tuples" and explain what each tuple can contain in the explanation below.

Reply:
@jorisvandenbossche thank you for taking a look!

I will try to rewrite the doc-comment and push again.